core/src/main/java/com/linecorp/armeria/internal/client/DefaultClientRequestContext.java

```diff
@@ -531,6 +528,14 @@ private DefaultClientRequestContext(DefaultClientRequestContext ctx,
 
         this.endpointGroup = endpointGroup;
         updateEndpoint(endpoint);
+        if (ctx.endpoint() == endpoint || endpoint == null) {
```

Comment:
Question) Is this compare by reference as opposed to equality intentionally?

Reply:
Originally, reference comparison was used to know the derivation was the initial attempt. The logic is changed to check the initial attempt more explicitly.

Reply:
>  reference comparison was used to know the derivation was the initial attempt

Are you suggesting that `ctx.endpoint() == endpoint` implies that it is the initial attempt?

Anyways, I was actually implying that it is possible that an endpoint group updates its endpoints, and comparing by reference won't reflect it. I think it's fine if you are aware of this

Reply:
`ctx.log.children().isEmpty()` would be enough but added reference equality for double check.
`ctx.endpoint() == endpoint && ctx.log.children().isEmpty()`

If a derivation is not an initial attempt, `acquireEventloop()` needs to be called even for the same `Endpoint` to update the internal state of `DefaultEventLoopScheduler`.